util/src/org/commcare/util/CommCareTransactionParserFactory.java

```diff
@@ -0,0 +1,189 @@
+package org.commcare.util;
+
+import java.io.IOException;
+
+import org.commcare.cases.ledger.Ledger;
+import org.commcare.cases.model.Case;
+import org.commcare.data.xml.TransactionParser;
+import org.commcare.data.xml.TransactionParserFactory;
+import org.commcare.util.mocks.MockUserDataSandbox;
+import org.commcare.util.mocks.UserXmlParser;
+import org.commcare.xml.CaseXmlParser;
+import org.commcare.xml.FixtureXmlParser;
+import org.commcare.xml.LedgerXmlParsers;
+import org.javarosa.core.model.instance.FormInstance;
+import org.javarosa.core.services.storage.IStorageUtilityIndexed;
+import org.javarosa.xml.util.InvalidStructureException;
+import org.javarosa.xml.util.UnfullfilledRequirementsException;
+import org.kxml2.io.KXmlParser;
+import org.xmlpull.v1.XmlPullParserException;
+
+/**
+ * A factory for covering all of the basic transactions expected in userspace
+ * against the mock data sandbox provided.
+ *
+ * This set of transactions should almost certainly be made uniform between
+ * different platform implementations, but the different platforms require some
+ * subtle difference right now which make that challenging.
+ *
+ * @author ctsims
+ */
+public class CommCareTransactionParserFactory implements TransactionParserFactory {
+
+    private TransactionParserFactory userParser;
+    private TransactionParserFactory caseParser;
+    private TransactionParserFactory stockParser;
+    private final TransactionParserFactory fixtureParser;
+
+    private final MockUserDataSandbox sandbox;
+
+    private int requests = 0;
+    private String syncToken;
+
+    public CommCareTransactionParserFactory(MockUserDataSandbox sandbox) {
+        this.sandbox = sandbox;
+
+        fixtureParser = new TransactionParserFactory() {
+            FixtureXmlParser created = null;
+
+            public TransactionParser getParser(KXmlParser parser) {
+                if (created == null) {
+                    created = new FixtureXmlParser(parser) {
+                        //TODO: store these on the file system instead of in DB?
+                        private IStorageUtilityIndexed<FormInstance> fixtureStorage;
+
+                        /*
+                         * (non-Javadoc)
+                         * @see org.commcare.xml.FixtureXmlParser#storage()
+                         */
+                        @Override
+                        public IStorageUtilityIndexed<FormInstance> storage() {
+                            if (fixtureStorage == null) {
+                                fixtureStorage = CommCareTransactionParserFactory.this.sandbox.getUserFixtureStorage();
+                            }
+                            return fixtureStorage;
+                        }
+                    };
+                }
+
+                return created;
+            }
+        };
+
+        this.initUserParser();
```

Comment:
Just legacy (IE: As mentioned in comments this was copied from Android, so it hadn't gotten moved yet).